src/detect-isdataat.c

```diff
@@ -56,11 +57,97 @@ static DetectParseRegex parse_regex;
 int DetectIsdataatSetup (DetectEngineCtx *, Signature *, const char *);
 #ifdef UNITTESTS
 static void DetectIsdataatRegisterTests(void);
+static void DetectAbsentRegisterTests(void);
```

Comment:
Why have you done entire `absent` lifecycle in the `detect-isdataat.c` file?

Reply:
because it used to be implemented as `isdataat:!0`

What do you think ?